core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -96,7 +128,56 @@ class DefaultAutoTopicCreationManager(
     requestContext: RequestContext
   ): Unit = {
     if (topics.nonEmpty) {
-      sendCreateTopicRequest(topics, Some(requestContext))
+      sendCreateTopicRequestWithErrorCaching(topics, Some(requestContext))
+    }
+  }
+
+  override def getTopicCreationErrors(
+    topicNames: Set[String],
+    errorCacheTtlMs: Long
+  ): Map[String, String] = {
+    // Proactively expire old entries using the provided TTL
+    expireOldEntries(errorCacheTtlMs)
+    
+    val errors = mutable.Map.empty[String, String]
+    
+    // Check requested topics  
+    topicNames.foreach { topicName =>
+      Option(topicCreationErrorCache.get(topicName)) match {
+        case Some(error) =>
+          errors.put(topicName, error.errorMessage)
+        case None =>
+      }
+    }
+    
+    errors.toMap
+  }
+
+  /**
+   * Remove expired entries from the cache using the provided TTL.
+   * Since we use LinkedHashMap with insertion order, we only need to check 
+   * entries from the beginning until we find a non-expired entry.
+   */
+  private def expireOldEntries(ttlMs: Long): Unit = {
```

Comment:
I think the expiration won't work if we have different TTLs for different groups, right?

Since the entries will not expire in insertion order.

I think we may need a priority queue and hashmap to solve this correctly. We should probably put this into a little helper class that is synchronized.